APIWrapper/ResponseParser.cs

```diff
@@ -43,8 +43,23 @@ public static ScheduleWeekContent ParseResponse(string myResponse)
                         var sc = new ScheduleHourContent(int.Parse(block["b"].ToString()), false,
                                                          lessonType, teacherName, roomName, subjectShortcut,
                                                          subjectName, SubjectType.None, new List<string>());
+
                         //-1 because API count day 1,2,3,4,5 and we store days at indexes 0,1,2,3,4
-                        weekTimetable.DaysInWeek[int.Parse(block["dw"].ToString()) - 1].BlocksInDay.Add(sc);
+                        var weekday = weekTimetable.DaysInWeek[int.Parse(block["dw"].ToString()) - 1];
+                        bool inserted = false;
+                        for (int curBlockIndex = 0; curBlockIndex < weekday.BlocksInDay.Count; curBlockIndex++)
+                        {
+                            if (weekday.BlocksInDay[curBlockIndex].IsSameBlockAs(sc))
+                            {
+                                weekday.BlocksInDay.Insert(curBlockIndex + 1, sc);
+                                inserted = true;
+                                break;
+                            }
+                        }
```

Comment:
Nice, but what about to create method to weekday - e.g.: GetSameBlock, which will return the same block, and if exits, you will join it, if not, you will add new one.
Then the code will be more readable, you will not to have think 15minuts, whoat for is this code block...

Reply:
I was confused a little, I would do it oposite way, not: hour search in day for same hour, but: day search for same hour:
weekday,GetIndexOfSameBlock(sc) 
But OK :-)
It would be good, when you fix some comment in review, to write there some message...